solr/core/src/java/org/apache/solr/schema/IntPointField.java

```diff
@@ -119,7 +121,16 @@ public Query getSetQuery(QParser parser, SchemaField field, Collection<String> e
       values[i] = parseIntFromUser(field.getName(), val);
       i++;
     }
-    return IntPoint.newSetQuery(field.getName(), values);
+    Query pointsQuery = IntPoint.newSetQuery(field.getName(), values);
```

Comment:
with the caveat that most of this is above my head, I think this code could be written in a simplified manner as
```
if (field.hasDocValues()) {
  return IntField.newSetQuery(field.getName(), values);
} else {
  return IntPoint.newSetQuery(field.getName(), values);
}
```
most of the new code seems to be the same as the IntField.newSetQuery method, so I am thinking calling into it could be a good idea. haven't checked the other types yet.